kind/README.adoc

```diff
@@ -0,0 +1,204 @@
+= How to create a Kubernetes cluster locally
+:icons: font
+:revdate: {docdate}
+:toc: left
+:toclevels: 2
+
+== Introduction
+
+The goal of this project is to simplify our life when we use kubernetes
+on a laptop. It currently supports the following features: 
+
+* Create/delete a K8s cluster using the latest cluster version, or the one you specify, 
+* Launch a local container registry exposed on port `localhost:5000`, 
+* Set up an Ingress
+controller to route the traffic. 
+
+The bash script has been developed supporting the installation of a docker registry according to the following options: 
+
+* Unsecure: plain HTTP
+* Secured: HTTP/HTTPS endpoints available
+
+== Prerequisite
+
+To be able to install Kind using this project the following tools are required.
+
+* jq: https://stedolan.github.io/jq/download/
+* Kind: https://github.com/kubernetes-sigs/kind/releases
+* Docker desktop: https://www.docker.com/products/docker-desktop
+* openssl
+
+== Usage
+
+From the repository root execute the following command.
+
+[source,bash]
+----
+./kind/deploy.sh command [parameters,...]
+----
+
+The available commands are:
+
+.Script commands
+[%header,cols="2m,4"]
+|===
+| Command | Meaning
+
+| install | Install the kind cluster
+| remove | Remove the kind cluster
+|===
+
+=== Relevant options
+
+Other relevant parameters are the following.
+
+.Script commands
+[%header,cols="2,5"]
+|===
+| Command | Meaning
+
+| `--ingress`
+
+[.fuchsia]#string# / [.red]#required# 
+a| Ingress to be deployed. 
+
+Choices: 
+
+* nginx
+* kourier
+
+| `--cluster-name`
+
+[.fuchsia]#string#
+a| Name of the cluster.
+
+Default: kind
+
+| `--port-map`
+
+[.fuchsia]#string#
+a| Comma separated list of ports to map on kind config.
+
+_e.g._: `ContainerPort1:HostPort1,...`
+
+[WARNING]
+====
+The ports required to be accessed from outside the kind cluster must be mapped between the host and the kind container using this parameter.
+====
+
+| `--provider`
+
+[.fuchsia]#string#
+a| Allows selecting the container runtime to be used. 
+
+Choices: 
+
+* *`docker` <= Default*
+* `podman`
+
+|===
+
+
+== Deploy
+
+Deploying a kind cluster is as simple as executing the following command.
+
+.kind deploy statement
+[source,bash]
+----
+curl -s -L "https://raw.githubusercontent.com/snowdrop/k8s-infra/main/kind/deploy.sh" | bash -s install --ingress nginx --cluster-name twm --provider podman --port-map "30000:30000,31000:31000"
+----
+
+=== Post deployment steps
+
+Allow using an insecure container registry.
+
+Add the `kind-registry.conf` file to the `/etc/containers/registries.conf.d/` folder having the following contents.
```

Comment:
**Notes**: This is correct for linux users but not for macos users and most probably if you use docker vs podman. To be reviewed please

Reply:
The [podman installation documentation](https://podman.io/getting-started/installation) doesn't provide information regarding the registry configuration for Mac OS. 

Do you have a `/etc/containers/registries.conf.d/` folder on your machine, @cmoulliard ?

```bash
$ ls -l /etc/containers/
total 8
drwxr-xr-x. 1 root root    0 Feb 14 14:42 certs.d
drwxr-xr-x. 1 root root   14 Feb 14 14:42 oci
-rw-r--r--. 1 root root  569 Feb 14 14:42 policy.json
-rw-r--r--. 1 root root 3897 Feb 14 14:42 registries.conf
drwxr-xr-x. 1 root root   74 Mar 30 12:57 registries.conf.d
drwxr-xr-x. 1 root root  132 Feb 20 15:37 registries.d
drwxr-xr-x  1 root root    0 Feb 14 14:42 systemd
```

Reply:
No as the Macos is not a Linux OS and by consequence do not support natively `containers`. This is why we need the help of docker (able to create locally a VM using Hyperkit).

Certificate generated are stored here:
```
ls -la /etc/docker/certs.d
total 0
drwxr-xr-x  7 root  wheel  224 Mar  8 15:02 .
drwxr-xr-x  3 root  wheel   96 Sep 17  2021 ..
drwxr-xr-x  3 root  wheel   96 Mar  8 15:02 127.0.0.1.sslip.io:5000
drwxr-xr-x  3 root  wheel   96 Mar  8 14:56 127.0.1.sslip.io:5000
drwxr-xr-x  3 root  wheel   96 Sep 17  2021 95.217.159.244:32500
drwxr-xr-x  4 root  wheel  128 Sep 28  2021 kind-registry:5000
drwxr-xr-x  3 root  wheel   96 Jun  2  2022 registry.harbor.192.168.1.90.nip.io
```
and registries here:
```
cat /Users/cmoullia/.docker/config.json
{
  "auths": {
    "ghcr.io": {},
    "https://index.docker.io/v1/": {},
    "index.docker.io": {},
    "localhost:5000": {},
    "quay.io": {},
  },
  "credsStore": "desktop"
}
```
**Remark**: There is also a `$HOME/.docker/certs.d` but I have no idea if it is used !!!

Reply:
I've separated the information for insecure registries for docker and podman.